### PR TITLE
try it

### PR DESCRIPTION
**Pull request by seanbb88**
Adding new class `GetGenres`
This gets the list of genres in the movie db. On log in we set this list in storage. 
When a user searches for a movie the hook pulls this list of genres and matches it with the returned movie so we can display it in the UI

### CODE REVIEW

File: flicks/views/movies.py [hunk 1]
```diff
@@ -5,25 +5,28 @@
...
         response = dict()
         statusCode = status.HTTP_200_OK
         try:
+            user_id = request.GET.get("userId")
+            user = User.objects.filter(id=user_id).first()
             movie_list = list()
-            movies = Movie.objects.all()
-            for movie in movies:
-                serializer = MovieSerializer(movie)
+            user_flick_movies =  UserFlick.objects.filter(user=user.id).select_related("movie")
```
Comment (seanbb88) [inline flicks/views/movies.py:19]: returning all flicks per user now

File: flicks/views/userFlick.py [hunk 2]
```diff
@@ -45,7 +45,17 @@ def put(self, request):
             user_id = data["userId"]
             movie_id = data["movieId"]
             watched = data["watched"]
-            UserFlick.objects.filter(movie=movie_id, user=user_id).update(watched=watched)
+            user_flick = UserFlick.objects.filter(movie=movie_id, user=user_id).update(watched=watched)
+
+            if user_flick:
+                # Update the watched attribute
+                user_flick.watched = watched
```
Comment (JonStults) [inline flicks/views/userFlick.py]: isn't this already happening in line 48

File: flicks/views/movieGenres.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        response = dict()
+        statusCode = status.HTTP_200_OK
+        try:
+            print('GET MOVIE GENRES')
+            url = "https://api.themoviedb.org/3/genre/movie/list?language=en"
+
+            headers = {
+            "accept": "application/json",
+            "Authorization": "Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIyNWE3MjkzMjg3MzA4MDY0OWU4M2ZmNzRhMDBhMjZkNiIsInN1YiI6IjY1NmZiODlmOTQ2MzE4MDEwMGM2YmE4ZiIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.37AzMiibDDo_bX1TMRmOeBJ-JwDHQFk1b8U9vEocq-Y"
+            }
```
Comment (JonStults) [inline flicks/views/movieGenres.py:32]: why is this token baked in?

Comment (seanbb88) [reply flicks/views/movieGenres.py:32]: Cause API_KEY wouldnt work for some reason although the documentation said it would.

Comment (seanbb88) [reply flicks/views/movieGenres.py:32]: I'm gonna fix this stuff when I get the Env deploy done

File: flicks/views/userFlick.py (hunk 2 above)
Comment (seanbb88) [reply flicks/views/userFlick.py]: its jsut checking to see if its in the db first before allowing the user to flip it to watched. . jasus

Comment (JonStults) [reply flicks/views/userFlick.py]: yeah but it's calling update so we should take that function off in line 48 if we wanna do this check first

Comment (seanbb88) [reply flicks/views/userFlick.py]: ahh i see what u mean

**Pull request merged**